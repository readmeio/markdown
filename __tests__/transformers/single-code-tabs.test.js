import { mdast } from '../../index';

describe('Single Code Tab Transformer', () => {
  it('wraps single code blocks with tabs if they have a lang or title set', () => {
    const md = `
\`\`\`javascript
const languageSet = true;
\`\`\`
`;

    const tree = mdast(md);
    expect(tree).toMatchSnapshot();
  });
});
