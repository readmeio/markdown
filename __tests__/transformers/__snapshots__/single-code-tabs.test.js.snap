// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Code Tab Transformer wraps single code blocks with tabs if they have a lang or title set 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "lang": "javascript",
          "meta": null,
          "type": "code",
          "value": "const languageSet = true;",
        },
      ],
      "className": "tabs",
      "data": Object {
        "hName": "code-tabs",
      },
      "type": "code-tabs",
    },
  ],
  "type": "root",
}
`;
